### ISSUE TITLE
Puppeteer as peer dependency?

### ISSUE THREAD

**Issue by jkroepke** [flood-io/element#132]
**Describe the bug**
Hi, the current version of puppeteer, file upload does not work: https://github.com/puppeteer/puppeteer/issues/5420

As workaround, puppeteer 2.0.0 should be use.

Since flood.io define the version of puppeteer I had to use a old version of flood.io which contains missing bug fixes.

It's like a dead lock for me.

If puppeteer is a peer dependency of flood.io I could define the puppeteer version itself while using the lastest version of flood.io.

**Comment by ivanvanderbyl** [comment]
Hi @jkroepke, we'll look into this, in the meantime, I'll upgrade puppeteer and test this out.

**Comment by huyennbl** [comment]
Hi @jkroepke, hope this solution can help you or you may use it as a workaround for current situation. 
Inside your element project folder
First, please remove your current yarn.lock or package-lock.json, 
Then define puppeteer 2.0.0 in package.json in both `dependencies` and `resolutions` 
It should look like this
```
{
	"name": "element-1",
	"version": "0.0.1",
	"description": "Flood Element test script",
	"private": true,
	"dependencies": {
		"@flood/element": "^1.0.8-stable.1",
		"prettier": "^1.10.2",
		"puppeteer": "2.0.0"
	},
	"resolutions": {
		"puppeteer": "2.0.0"
	}
}
```

Then `yarn install` and it should work
FYI: https://classic.yarnpkg.com/en/docs/selective-version-resolutions/

**Comment by jkroepke** [comment]
Hi @huyennbl 

i know. But's its yarn exclusive. We have to stay on `npm`.

**Comment by huyennbl** [comment]
Ok, so I got the idea :D

**Comment by huyennbl** [comment]
@jkroepke hey please try again with npm, the solution works on my machine. So I think this one is kind of package.json standard, not yarn exclusive
Screenshot from my package-lock.json
![image](https://user-images.githubusercontent.com/33422297/80951150-7ff44000-8e21-11ea-92c6-8c320d7d2250.png)
node: 12.16.1, npm: 6.13.4

**Comment by jkroepke** [comment]
Did you use?

https://github.com/rogeriochaves/npm-force-resolutions

**Comment by huyennbl** [comment]
I don't use the `npm-force-resolutions` as you mentioned. 
Yeah I checked again and `resolutions` is not a part of package.json. Not sure why it installs version 2.0.0 for me. 
Which version of puppetter is installed on your machine using npm?  2.1.1 or 2.0.0

**Comment by jkroepke** [comment]
It is still `2.0.0` if you remove `package-lock.json` and `node_modules`

**Comment by huyennbl** [comment]
Yes it is still 2.0.0
I've just also tried on another computer (with npm of course), and still 2.0.0 :D

**Comment by jkroepke** [comment]
I got it.

You got `puppeteer@2.0.0` because you define it in your `dependencies`. 

You should recheck your `package-lock.json`, you should have a 2nd puppeteer dependency under `@flood/element-cli` like `/node_modules/@flood/element-core/node_modules/puppeteer` and check the version of this.

**Comment by huyennbl** [comment]
Yes I know about that there are nested puppeteer dependencies inside other subpackages e.g. @flood/element-cli. But the version in the 'root' dependencies will be the one that npm install inside node_modules. So I expect element will run with this version (please correct if I'm getting this wrong).
From `node_modules/puppeteer/package.json`
![image](https://user-images.githubusercontent.com/33422297/80954341-66ee8d80-8e27-11ea-9974-53d9f0c6cc92.png)

**Comment by jkroepke** [comment]
Still same results if you are using
```
"@flood/element": "^1.2.3",
```

instead

```
"@flood/element": "^1.0.8-stable.1",
```

**Comment by huyennbl** [comment]
`node_modules/puppeteer/package.json` is still showing version 2.0.0 when I retried with element 1.2.3 (I used `npm install --no-cache` to ensure no caching problem here)
My full package.json in case you need for reference
```
{
	"name": "element-1",
	"version": "0.0.1",
	"description": "Flood Element test script",
	"private": true,
	"prettier": {
		"semi": false,
		"singleQuote": true,
		"trailingComma": "all",
		"printWidth": 100,
		"useTabs": true,
		"tabWidth": 2,
		"bracketSpacing": true,
		"jsxBracketSameLine": true,
		"arrowParens": "avoid"
	},
	"dependencies": {
		"@flood/element": "^1.2.3",
		"prettier": "^1.10.2",
		"puppeteer": "2.0.0"
	},
	"resolutions": {
		"puppeteer": "2.0.0"
	}
}

```

**Comment by jkroepke** [comment]
Your example results into this:

```
jan@JOK-MAC r % npm ls puppeteer
element-1@0.0.1 /Users/jan/Downloads/r
├─┬ @flood/element@1.2.3
│ └─┬ @flood/element-core@1.2.3
│   └── puppeteer@2.1.1
└── puppeteer@2.0.0
```

After removing puppeteer from the top level dependencies, I got this:

```
jan@JOK-MAC r % npm ls puppeteer
element-1@0.0.1 /Users/jan/Downloads/r
└─┬ @flood/element@1.2.3
  └─┬ @flood/element-core@1.2.3
    └── puppeteer@2.1.1
```

Whatever, such feature like `resolutions` not documented and does not exists on `npm`. It a planned feature for npm v7.

**Comment by jkroepke** [comment]
Hi @ivanvanderbyl ,

do you have an update for us?
